### PR TITLE
Fullfill project requirements

### PR DESCRIPTION
**Pull request by koomekevo**
This Pull Request has been created to fulfill the project requirements as shown below. 

#### Design
- You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
  - Colors.
  - Typography: font face, size and weight.
  - Layout: composition and space between elements.

> NOTE: In these design guidelines there are several UIs that you won't need for this exercise; also, some pages are not given a design and you will create them following the design guidelines of the other pages.
Original design idea by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella).

The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

#### Interactions
- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (catagories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.

#### Technical requirements

- You should use Postgres as your database.
- You should use devise for authentication.
- You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
- You can use a view template engine of your choice (.erb, .slim, .haml).
- The project should be deployed and accessible online.
- Your database schema should reflect the following structure:

> NOTE: do not change column names visible in the diagram. You need to change the "Entity"  name according to the theme you have chosen for your project (please note: "Transaction" is a name already used by ActiveRecord, so using it as a name for your model and table will result in an error).

<p align="center">  
rails_cash_erd_diagram
</p>

### CODE REVIEW

File: app/views/splashes/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<div>
+  <div class="p-5">  
+    <h1 class="logo-text text-center py-5">RailsCash</h1>
+  </div>
+  <div class="d-grid">
+    <%= link_to "Log in", new_user_session_path, :class => "btn btn-primary" %><br>
+    <%= link_to "Sign up", new_user_registration_path, :class => "btn btn-light" %>
+  </div>
+</div>
```
Comment (mmsesay) [inline app/views/splashes/index.html.erb]: - [ ] You did well coming up with this project. However, I will suggest you try to follow the guideline [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=) in terms of :
    - Font size
    - Colors
    - Font family etc.

The font family should be : `Proxima Nova`
The primary blue colour should be `#3778C2` not `#0D6EFD`
The primary green colour should be `#5FB523`

File: app/views/entities/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        <p class="card-text"><%= entity.amount %></p>
+        <p class="card-text"><small class="text-muted"><%= entity.created_at %></small></p>
+      </div>
+    </div>
+  </div>
+<% end %>
+
+<div class="p-2">
+  <%= link_to "Add transaction", new_group_entity_path(@group, @entity), :class => "btn btn-primary" %>
+</div>
```
Comment (mmsesay) [inline app/views/entities/index.html.erb]: - [ ] Just like I mentioned above, please try to follow the design. It is a very good practice. Please note that the scan button is the add transaction button in your design.

Your Transaction page | Design Guide |
:------------------------------:|:-----------------:|
a6| a4

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @koomekevo,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Verifying that your tests are working fine as expected. I ran into some problems when I tried to run them. Maybe, it might be my system, but please just take the time to verify.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/views/entities/index.html.erb (hunk 2 above)
Comment (peterrobert) [reply app/views/entities/index.html.erb]: This change has been done well

File: app/views/splashes/index.html.erb (hunk 1 above)
Comment (peterrobert) [reply app/views/splashes/index.html.erb]: This change has been done well

**Review by peterrobert — APPROVED** [review]
Hi @koomekevo 
, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**